File: src/canadianseh.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	line := scanner.Text()
	if strings.HasSuffix(line, "eh?") {
		fmt.Println("Canadian!")
	} else {
		fmt.Println("Imposter!")
	}
}

File: src/canadianseh.java
import java.util.Scanner;

class CanadiansEh {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        String line = s.nextLine();
        if (line.endsWith("eh?")) {
            System.out.println("Canadian!");
        } else {
            System.out.println("Imposter!");
        }
    }
}
